### PR TITLE
Use the official Ubuntu repository for Visual Studio Code

### PR DESCRIPTION
**Pull request by mseri**
Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

### CODE REVIEW

**Comment by Frezzle** [comment]
Is there a difference from the other one? Is there a reason @jonludlam didn't use this initially?

**Comment by mseri** [comment]
This version allows you to always have the latest version just running `apt update && apt upgrade` without having to manually re-download the deb file. I would not be surprised if key and repo could be easily added with `apt-add`, on my workstation I did it by hand as above (and as described on vscode linux page: https://code.visualstudio.com/docs/setup/linux).

It may be that it was not there because it is a recent(-ish) feature. Until few months ago you could only download the latest deb by hand.

File: Vagrantfile [hunk 1]
```diff
@@ -36,10 +36,13 @@ Vagrant.configure("2") do |config|
     sudo apt-get install xubuntu-desktop gksu leafpad synaptic
     sudo add-apt-repository -y "ppa:webupd8team/sublime-text-3"
     sudo apt-add-repository -y "ppa:webupd8team/atom"
+    curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > microsoft.gpg
+    sudo mv microsoft.gpg /etc/apt/trusted.gpg.d/microsoft.gpg
+    sudo sh -c 'echo "deb [arch=amd64] https://packages.microsoft.com/repos/vscode stable main" > /etc/apt/sources.list.d/vscode.list'
```
Comment (gaborigloi) [inline Vagrantfile:41]: Could we also do `sudo echo "deb [arch=amd64] https://packages.microsoft.com/repos/vscode stable main" > /etc/apt/sources.list.d/vscode.list` without the `sh -c`?

**Review by gaborigloi — APPROVED** [review]
Looks good, just one minor comment about a possible simplification. This method install the exact same version of `code` as the old one, `1.12.1-1493934083`. (I've tested it in a Docker container and it worked.)

**Comment by gaborigloi** [comment]
One question is whether we need to install this:
```
sudo apt-get install apt-transport-https
```
to make `apt-get update` work with the newly added repository.

**Review by gaborigloi — APPROVED** [review]
Looks good, just one minor comment above. I've tested it with `vagrant up` and `vagrant provision`, and `code` was installed successfully. (After that it actually failed to build xapi, but that's not relevant to this PR.) It seems that `apt-transport-https` is installed in the box.

File: Vagrantfile (hunk 1 above)
Comment (mseri) [reply Vagrantfile:41]: I used the way suggested by Microsoft on the official page: https://code.visualstudio.com/docs/setup/linux

Comment (gaborigloi) [inline Vagrantfile]: These are pated from the official Microsoft instructions, let's stay as close to them as possible.

**Pull request merged**